### PR TITLE
update: add prompts and retriever chain

### PR DESCRIPTION
**Pull request by Znis**
This pull request does the following things:

- Adds a retriever chain
- Adds the necessary util functions for the retriever chain
- Adds the prompts
- Adds the models
- Changes the config file, embedding script, requirements file slightly

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,10 +1,11 @@
 loguru==0.7.2
-langchain==0.2.16
-langchain-community==0.2.16
-langchain-openai==0.1.25
+langchain==0.3.0
```
Comment (amunamun) [inline requirements.txt:2]: While changing the version make sure it doesnt break anything from previous code

File: nepal_constitution_ai/retriever/utils.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    Returns:
+        OpenaiModel: The selected OpenAI model.
+
+    Raises:
+        ValueError: If the provided LLM name is invalid.
+    """
+    if llm_name.lower() in settings.OPENAI_MODEL:
+        llm_model = OpenaiModel(llm_name).model_selection()
+    else:
+        return ValueError("Wrong llm model selected")
```
Comment (amunamun) [inline nepal_constitution_ai/retriever/utils.py]: its preferred that u use try catch if you are returning any Errors. Also, log this

File: nepal_constitution_ai/retriever/utils.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+        return PineconeVectorStore(
+            index_name=pinecone_index,
+            embedding=embedding,
+            pinecone_api_key= settings.PINECONE_API_KEY,
+            
+        ).as_retriever(search_kwargs={"k": k})
+
+    else:
+        raise ValueError(f"Unsupported vector database: {vector_db}")
```
Comment (amunamun) [inline nepal_constitution_ai/retriever/utils.py:59]: same here

**Review by kalexrt — COMMENTED** [review]
looks good

File: requirements.txt (hunk 1 above)
Comment (kalexrt) [reply requirements.txt:2]: Resolved in new commit

File: nepal_constitution_ai/retriever/utils.py (hunk 3 above)
Comment (kalexrt) [reply nepal_constitution_ai/retriever/utils.py:59]: Resolved in new commit

File: nepal_constitution_ai/retriever/utils.py (hunk 2 above)
Comment (kalexrt) [reply nepal_constitution_ai/retriever/utils.py]: Resolved in new commit

**Review by kalexrt — APPROVED** [review]
approved

**Pull request merged**